src/wp-includes/class-wp-textdomain-registry.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * Locale API: WP_Textdomain_Registry class
+ *
+ * @package WordPress
+ * @subpackage i18n
+ * @since 6.1.0
+ */
+
+/**
+ * Core class used for registering text domains.
+ *
+ * @since 6.1.0
+ */
+class WP_Textdomain_Registry {
+	/**
+	 * List of domains and their language directory paths.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @var array
+	 */
+	protected $domains = array();
+
+	/**
+	 * Holds a cached list of available .mo files to improve performance.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @var array
+	 */
+	protected $cached_mo_files;
+
+	/**
+	 * Returns the MO file path for a specific domain and locale.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $domain Text domain.
+	 * @param string $locale Locale.
+	 *
+	 * @return string|false MO file path or false if there is none available.
+	 */
+	public function get( $domain, $locale ) {
+		if ( isset( $this->domains[ $domain ][ $locale ] ) ) {
+			return $this->domains[ $domain ][ $locale ];
+		}
+
+		return $this->get_path_from_lang_dir( $domain, $locale );
+	}
+
+	/**
+	 * Determines whether any MO file paths are available for the domain.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $domain Text domain.
+	 * @return bool Whether any MO file paths are available for the domain.
+	 */
+	public function has( $domain ) {
+		return ! empty( $this->domains[ $domain ] );
+	}
+
+	/**
+	 * Returns the current (most recent) MO file path for a specific domain.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $domain Text domain.
+	 * @return string|false Current MO file path or false if there is none available.
+	 */
+	public function get_current( $domain ) {
+		if ( isset( $this->domains[ $domain ]['current'] ) ) {
+			return $this->domains[ $domain ]['current'];
+		}
+
+		return false;
+	}
+
+
+	/**
+	 * Sets the MO file path for a specific domain and locale.
+	 *
+	 * Also sets the 'current' property for direct access
+	 * to the path for the current (most recent) locale.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string       $domain Text domain.
+	 * @param string       $locale Locale.
+	 * @param string|false $path Language directory path or false if there is none available.
+	 */
+	public function set( $domain, $locale, $path ) {
+		$this->domains[ $domain ][ $locale ] = $path ? trailingslashit( $path ) : false;
+		$this->domains[ $domain ]['current'] = $this->domains[ $domain ][ $locale ];
```

Comment:
Would it make sense to actually have a `current` property here? It's highly unlikely that we'll ever have a locale named `current` but I think it makes things a bit more clear.

```suggestion
		$this->current[ $domain ] = $this->domains[ $domain ][ $locale ];
```

Reply:
Yeah I thought about that too.

Maybe then worth renaming the`domains` property to `all`? So we have `all` and `current`?